src/compute/src/logging/persist.rs

```diff
@@ -0,0 +1,146 @@
+use std::collections::HashMap;
+use std::time::Duration;
+
+use differential_dataflow::operators::Consolidate;
+use differential_dataflow::{AsCollection, Collection};
+use mz_persist_client::read::ListenEvent;
+use timely::dataflow::channels::pact::Exchange;
+use timely::dataflow::operators::generic::builder_rc::OperatorBuilder;
+use timely::dataflow::operators::{Event, ToStreamAsync};
+use timely::dataflow::Scope;
+
+use mz_persist_client::{PersistClient, PersistLocation, ShardId};
+use mz_repr::{Diff, Row, Timestamp};
+use mz_timely_util::operators_async_ext::OperatorBuilderExt;
+
+pub(crate) fn persist_roundtrip<G>(
+    collection: Collection<G, (Row, Row), Diff>,
+) -> Collection<G, (Row, Row), Diff>
+where
+    G: Scope<Timestamp = Timestamp>,
+{
+    let scope = collection.scope();
+    let active_worker_index = 0;
+    let timeout = Duration::from_secs(5);
+
+    let (write, read) = if active_worker_index == scope.index() {
+        let persist_location = PersistLocation {
+            consensus_uri: "sqlite:///tmp/mz-persist/consensus".into(),
+            blob_uri: "file:///tmp/mz-persist/blob".into(),
+        };
+        let shard_id = ShardId::new();
+
+        let (blob, consensus) = futures_executor::block_on(persist_location.open(timeout))
+            .expect("cannot open persist location");
+        let persist_client =
+            futures_executor::block_on(PersistClient::new(timeout, blob, consensus))
+                .expect("cannot open client");
+        let (write, read) = futures_executor::block_on(
+            persist_client.open::<Row, Row, Timestamp, Diff>(timeout, shard_id),
+        )
+        .expect("could not open persist shard");
+        (Some(write), Some(read))
+    } else {
+        (None, None)
+    };
+
+    // Ensure the collection contains no 0 diffs, as persist chokes on those.
+    // Aside from that, it seems to be a good idea to reduce the data as much
+    // as possible before sending it over the network.
+    let collection = collection.consolidate();
```

Comment:
consolidating is very expensive. It's better to use `consolidate_stream`, or make it a requirement to the caller that there are no 0 diffs.